staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -699,12 +699,17 @@ var (
 // priority, there are three factors affect whether to add/remove the
 // FinalizerOrphanDependents: options, existing finalizers of the object,
 // and e.DeleteStrategy.DefaultGarbageCollectionPolicy.
-func shouldOrphanDependents(e *Store, accessor metav1.Object, options *metav1.DeleteOptions) bool {
-	if gcStrategy, ok := e.DeleteStrategy.(rest.GarbageCollectionDeleteStrategy); ok {
-		if gcStrategy.DefaultGarbageCollectionPolicy() == rest.Unsupported {
-			// return  false to indicate that we should NOT orphan
-			return false
-		}
+func shouldOrphanDependents(ctx genericapirequest.Context, e *Store, accessor metav1.Object, options *metav1.DeleteOptions) bool {
+	// Get default GC policy from this REST object type
+	gcStrategy, ok := e.DeleteStrategy.(rest.GarbageCollectionDeleteStrategy)
```

Comment:
Why do you need to change these lines? It seems you only need to pass `ctx` to `DefaultGarbageCollectionPolicy()`?

Reply:
`DefaultGarbageCollectionPolicy` is called twice in L704 and L737.

I think we should call it once and get the value.